master/pkg/tasks/task.go

```diff
@@ -119,7 +120,17 @@ func (t TaskSpec) EnvVars() map[string]string {
 		"DET_CONTAINER_ID":  t.ContainerID,
 		"DET_SESSION_TOKEN": t.AllocationSessionToken,
 		"DET_USER_TOKEN":    t.UserSessionToken,
-		"DET_USER":          t.Owner.Username,
+		"DET_WORKDIR":       t.WorkDir,
+	}
+	if t.Owner != nil {
+		e["DET_USER"] = t.Owner.Username
+	}
+
+	if t.TaskContainerDefaults.NCCLPortRange != "" {
+		e["NCCL_PORT_RANGE"] = t.TaskContainerDefaults.NCCLPortRange
+	}
```

Comment:
can we just drop this?

We once had a fork of NCCL that supported setting port ranges, but we no longer maintain that fork or ship it to users

It's basically an ignored config field now.

(the GLOO_PORT_RANGE is built into gloo, so it's still valid).

Reply:
This allows users to override the GLOO port range if they are hitting problems.
With Slurm+Singularity the ports are global on the compute nodes for all experiments.   If more than one experiment is running on the same node there can be conflicts.

Reply:
So I'm not sure what's up here.      

chore: remove legacy code/docs for NCCL/Gloo port range config (#4187)

Which removed it, but @stoksc said earlier in this review to change it from:

`e["GLOO_PORT_RANGE"] = t.TaskContainerDefaults.NCCLPortRange`

to 

`e["GLOO_PORT_RANGE"] = t.TaskContainerDefaults.GLOOPortRange`

I guess I can just drop it, unless  @stoksc was really trying to put it back.

Reply:
Let's just leave it for now and I can PR to remove the config entirely from the master shortly?

Reply:
I think removing it from the config is technically a breaking feature, which is why it was still allowed in the config even after #4187.  

But `NCCL_PORT_RANGE` is an environment variable that [I personally added to our nccl fork three years ago](https://github.com/determined-ai/nccl/commit/8768cb3f881ce198825294b62a435e3f3a804baf).  That fork is no longer used anywhere, and upstream nccl doesn't use `NCCL_PORT_RANGE`, so setting `NCCL_PORT_RANGE` will have no effect (except maybe confusing a user somewhere).

I'm not talking about removing the GLOO_PORT_RANGE environment variable, only the NCCL_PORT_RANGE.